mlblocks/mlpipeline.py

```diff
@@ -154,6 +154,45 @@ def _get_block_outputs(self, block_name):
 
         return outputs
 
+    def _get_block_outputs_dict(self, block_name):
+        """Get dictionary of output variables for the given block."""
+        block = self.blocks[block_name]
+        outputs = deepcopy(block.produce_output)
+        output_names = self.output_names.get(block_name, dict())
+        output_dict = {}
+        for output in outputs:
+            name = output['name']
+            context_name = output_names.get(name, name)
+            output_dict[context_name] = output
+
+        return output_dict
+
+    def _get_block_inputs_dict(self, block_name):
+        """Get dictionary of input variables for the given block."""
+        block = self.blocks[block_name]
+        print(block.produce_args)
+        inputs = deepcopy(block.produce_args)
+        input_names = self.input_names.get(block_name, dict())
+        inputs_dict = {}
+        for input_value in inputs:
+            name = input_value['name']
+            context_name = input_names.get(name, name)
+            inputs_dict[context_name] = input_value
+        return inputs_dict
```

Comment:
Let's always add a black line before an indentation decrease, to separate the different logical blocks.